eng/common/TestResources/New-TestResources.ps1

```diff
@@ -625,8 +651,8 @@ try {
         $userAccountName = $userAccount.UserPrincipalName
         Log "User authentication with user '$userAccountName' ('$TestApplicationId') will be used."
     }
-    # If no test application ID was specified during an interactive session, create a new service principal.
-    elseif (!$CI -and !$TestApplicationId) {
+    # If user has specified -ServicePrincipalAuth
+    elseif (!$CI -and $ServicePrincipalAuth) {
```

Comment:
Seems this needs to require `$TestApplicationId`, then. PowerShell does support a rich set of APIs for validating parameters if you want to know more. It's more idiomatic and does positively affect the CLI.

Reply:
We changed behavior in the script to default to the `-UserAuth` behavior of today. `-ServicePrinipalAuth` creates a service principal on behalf of the user for testing. 

Perhaps this should be something more like `-CreateServicePrincipal`

Either way `TestApplicationId` would be ignored.

Reply:
We kept the name `-ServicePrincipalAuth` but changed the behaviors a bit. We now use `ServicePrincipalAuth` to determine whether to log in as the provisioner app optionally supplied in `-ProvisionerApplication___` or to create a service principal on behalf of the user if no provisioner app info is supplied. 

The `-UserAuth` param was removed and it's now the default behavior.